image/image_test.go

```diff
@@ -51,3 +53,38 @@ func TestMarshalKeyOrder(t *testing.T) {
 		t.Fatal("invalid key order in JSON: ", string(b))
 	}
 }
+
+func TestNewChildImage(t *testing.T) {
```

Comment:
Yes, I agree this is very important. I think we've lost track of what most tests are supposed to be testing.

I really prefer when a test describes the functionality and the test case instead of being too specific about a previous bug. Bugs are hard to describe and aren't really meaningful in many context, but features or conditions should retain their meaning.

I'm also a fan of keeping the description in the test name. The format I try to use for test names is:

```
Test<Unit name><Condition>()
```

Where unit name is the function name or struct and function name, and the condition is the branch within the function being tested. I think this format makes it really easy to identify what is being tested.

I've updated the test name to be `TestNewChildImageFromImageWithRootFS` which I think is an accurate description of this test. It's not really a test specific to the bug as it tests more than just a single field, but it does fail without the fix, so still works as a regression test.